packages/admin/admin/src/translator/TranslationDialog.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle, Grid, InputBase } from "@mui/material";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { messages } from "../messages";
+
+interface TranslationDialogProps {
+    open: boolean;
+    close: () => void;
+    originalText: string;
+    translatedText: string;
+    applyTranslation: (value: string) => void;
+}
+
+export const TranslationDialog: React.FC<TranslationDialogProps> = (props) => {
+    const { open, close, originalText, translatedText, applyTranslation } = props;
+
+    const [translation, setTranslation] = React.useState<string>(translatedText);
```

Comment:
Wouldn't it be nice if the RTE would work the same way and have a local state?

Reply:
OK, managed to refactor the TranslationDialog component when checking this 004a2e569a875c6a917b3b22c4e4a8138d46541f